TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -282,31 +283,187 @@ public virtual DataType PrimitiveDataType
+                            }
+                            else
+                            {
+                                m = 4;
+                            }
+                            break;
+                        case DataUsage.Index:
+                        case DataUsage.Pointer:
+                        case DataUsage.ProcedurePointer:
+                        case DataUsage.ObjectReference:
+                        case DataUsage.FunctionPointer:
+                        case DataUsage.PackedDecimal:
+                            m = 4; 
+                            break;
+                        case DataUsage.FloatingPoint:
+                            m = 8;
+                            break;
+                    }
+
+                    if (occupiedMemory % m > 0)
+                    {
+                        _slackBytes = m - (occupiedMemory % m);
+                    }
+
+
+                }
+
+                return _slackBytes.Value;
+
+            }
+        }
+
+        private long? _startPosition = null;
+        public virtual long StartPosition
+        {
+            get
+            {
+                if (_startPosition.HasValue)
+                {
+                    return _startPosition.Value;
+                }
+
+                if (this is DataRedefines)
+                {
+                    SymbolReference redefined = ((DataRedefinesEntry)CodeElement)?.RedefinesDataName;
+                    var results = SymbolTable.GetVariables(redefined);
+
+                    if (results.Count() == 1)
+                    {
+                        _startPosition = results.First().StartPosition;
+                        return _startPosition.Value;
+                    }
+                    
+                }
+
+                if (Parent is DataSection)
+                {
+                    _startPosition = 1;
                 }
                 else
-                    return 0;
+                {
+                    for (int i = 0; i < Parent.Children.Count; i++)
+                    {
+                        Node sibling = Parent.Children[i];
+                        
+                        if (i == Parent.ChildIndex(this) - 1)
+                        {
+                            while(sibling is DataRedefines)
+                                sibling = Parent.Children[i - 1];
+
+                            _startPosition = ((DataDefinition)sibling).PhysicalPosition + 1 + SlackBytes;
+                        }
+                            
+                    }
+                    if (_startPosition == null)
+                    {
+                        _startPosition = (Parent as DataDefinition)?.StartPosition;
+                    }
+                }
 
-                return _length.Value;
+                return _startPosition ?? 0;
             }
         }
 
+        public virtual long PhysicalPosition => StartPosition + PhysicalLength - 1 + SlackBytes;
```

Comment:
Can you add comment, what is the difference between StartPosition and PhysicalPosition?